src/main/java/seedu/address/model/leave/Status.java

```diff
@@ -29,6 +31,9 @@ private Status(StatusType status) {
      */
     public static Status of(String status) throws IllegalArgumentException {
         requireNonNull(status);
+        if (!status.matches("PENDING|APPROVED|REJECTED")) {
```

Comment:
Are we enforcing uppercase?

Reply:
We can remove it at a later date the strict enforcement, as of now the reason it is as such is because the underneath implementation matches for the enum value